### ISSUE TITLE
Isolate history between instances [for sqlite history]

### ISSUE THREAD

**Issue by Hofer-Julian** [nushell/nushell#4981]
### Related problem

I've noticed that nushell synchronizes history between open instances. While that sounds nice, I found it less useful in practice.

I often have two terminals open where  I run the same command multiple times. Think `cargo run --bin one` in the first terminal and `cargo run --bin two` in the second one.

In another shell I would just press up and enter to run the correct command again. With nushell I have to potentially press up as often as necessary until the correct command pops up again. 

### Describe the solution you'd like

Isolate the history between instances and only merge them after one of the instances is closed. 

### Describe alternatives you've considered

Add an option to isolate instances

**Comment by Hofer-Julian** [comment]
Correction: I've tried it again, and you get the history of terminal 2 in terminal 1 only after executing at least one command in terminal 1.
Meaning repeating the same command in a separate terminal actually works.

**Comment by sholderbach** [comment]
I am gonna leave that open for now to provide the option to not do history file IO with every enter press or resemble the behavior of the archaic shells. But things might change in this space as we are looking into a bit fancier history (sqlite backed to have some metadata)

**Comment by sholderbach** [comment]
There is currently a config option that can turnoff the syncing for every enter so the sessions behave separate. So updates will only happen upon closing a session. 

https://github.com/nushell/nushell/blob/60b58630588e7ec0085a8d6e4fffe56d98ba767a/docs/sample_config/default_config.nu#L146

But this should be superseded by the more options with the database history

**Comment by devurandom** [comment]
`sync_history_on_enter: false` does not work with `history_file_format: "sqlite"`.

**Comment by sholderbach** [comment]
Right we currently do not take the session IDs stored with SQlite into account.

**Comment by mrbarletta** [comment]
I have this edge case as well.
I `ssh`  to a server, logout and the last command from nushell comes from another instance.

**Flow:**
`ssh me@server.com`
`reboot` ( I got kicked out)
key up to log in again `some other command`

**Comment by sholderbach** [comment]
Intended to be closed by #9063 (remaining issues there)

**Comment by Hofer-Julian** [comment]
I think so, too.
